Generator/src/main/java/com/microsoft/fluentui/generator/Extensions.kt

```diff
@@ -0,0 +1,161 @@
+import com.microsoft.fluentui.generator.model.resources.Resource
+import com.microsoft.fluentui.generator.model.resources.ResourceType
+import com.microsoft.fluentui.generator.model.resources.STRING_PATTERN
+import java.io.File
+import java.util.*
+
+
+fun String.writeToResourceFile(path: String, contentDescription: String) {
+    File(path).apply {
+        if (exists() && !readText().contains(AUTOGENERATED_MESSAGE, ignoreCase = true)) {
+            printError("File $path already exists")
+        } else {
+            parentFile.mkdirs()
+            writeText(this@writeToResourceFile)
+            println("$contentDescription -> ${this.path}")
+            global_resourceCount++
+
+            if (!this@writeToResourceFile.contains(AUTOGENERATED_MESSAGE, ignoreCase = true)) {
+                printWarning("File ${this.path} does not contain the autogenerated header")
+            }
+        }
+    }
+}
+
+fun String.writeToKotlinFile(path: String, contentDescription: String) {
+    File(path).apply {
+        if (exists() && !readText().contains(AUTOGENERATED_MESSAGE, ignoreCase = true)) {
+            printError("File $path already exists")
+        } else {
+            parentFile.mkdirs()
+            writeText(this@writeToKotlinFile)
+            println("$contentDescription -> ${this.path}")
+            global_resourceCount++
+
+            if (!this@writeToKotlinFile.contains(AUTOGENERATED_MESSAGE, ignoreCase = true)) {
+                printWarning("File ${this.path} does not contain the autogenerated header")
+            }
+        }
+    }
+}
+
+fun String.resolveResourceReference(resourceType: ResourceType) = when (this[0]) {
+    '$' -> "@${resourceType.resourceName}/".plus(toAndroidReference(resourceType))
+    else -> this
+}
+
+fun String.autoResolveReference() = when {
+    this.startsWith('$') -> {
+        when {
+            contains("color", ignoreCase = true) -> resolveResourceReference(ResourceType.COLOR)
+            contains("typography", ignoreCase = true) -> resolveResourceReference(ResourceType.STYLE)
+            isValidMetricRef() -> {
+                when (global_getResourceType(toAndroidReference().decapitalize())) {
+                    ResourceType.DIMENSION -> resolveResourceReference(ResourceType.DIMENSION)
+                    ResourceType.INTEGER-> resolveResourceReference(ResourceType.INTEGER)
+                    ResourceType.FLOAT-> resolveResourceReference(ResourceType.DIMENSION)
+                    null -> { printWarning("Unknown metric reference \"$this\""); this }
+                    else -> { printWarning("Unsupported metric type for \"$this\""); this }
+                }
+            }
+            else -> this
+        }
+    }
+    this.startsWith('#') -> {
+        if (length == 9) {
+            // alpha value in android is at the beginning #AARRGGBB
+            "#" + this.substring(7) + this.substring(1, 7)
+        } else this
+    }
```

Comment:
Will alpha values be applied to all Fluent colors?

Reply:
Nope, only for the values which have mentioned alpha in the end
In yml opacity is defined as opacity14: "#00000024"
So this gets converted to #2400000
For colors like grey10: "#1A1A1A" it is declared directly